### PR TITLE
Add key vault for auto-unseal

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/5436

### CODE REVIEW

File: modules/azure/vault/vault.tf [hunk 1]
```diff
@@ -14,6 +14,11 @@ resource "azurerm_virtual_machine" "vault" {
   network_interface_ids = ["${var.network_interface_ids[0]}"]
   vm_size               = "${var.vm_size}"
 
+  identity =
+  {
+    type = "SystemAssigned"
+  }
```
Comment (kopiczko) [inline modules/azure/vault/vault.tf]: Will it work on german cloud (if we still support it)?

Comment (corest) [reply modules/azure/vault/vault.tf]: Nope. But I'll add conditionals for aws china and azure german/stack in separate PR

**Comment by corest** [comment]
Important docs update on requirements of the `terraform` AD group added. @giantswarm/sig-infrastructure ptal

**Comment by corest** [comment]
Merging as destroy takes ages on azure

**Pull request merged**